### PR TITLE
feat(frontend): expose compact task progress via SQL

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

as title

https://github.com/risingwavelabs/risingwave/pull/13051#pullrequestreview-1736298084

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by Li0k** [comment]
image

File: src/ctl/src/cmd_impl/hummock/compaction_group.rs [hunk 1]
```diff
@@ -129,107 +127,6 @@ pub async fn split_compaction_group(
     Ok(())
 }
 
-pub async fn list_compaction_status(context: &CtlContext, verbose: bool) -> anyhow::Result<()> {
```
Comment (hzxa21) [inline src/ctl/src/cmd_impl/hummock/compaction_group.rs:132]: How about keeping the risectl command?

File: src/frontend/src/meta_client.rs [hunk 2]
```diff
@@ -246,11 +249,15 @@ impl FrontendMetaClient for FrontendMetaClientImpl {
         self.0.list_hummock_meta_config().await
     }
 
-    async fn list_compact_task_assignment(&self) -> Result<Vec<CompactTaskAssignment>> {
+    async fn list_compaction_status(&self) -> Result<Vec<CompactTaskAssignment>> {
```
Comment (hzxa21) [inline src/frontend/src/meta_client.rs:252]: Given that the RPC is ListCompactTaskAssignment and the returning type is CompactTaskAssignment, I think it is better to keep the original function name.

File: proto/hummock.proto [hunk 3]
```diff
@@ -706,6 +699,12 @@ message RiseCtlListCompactTaskAssignmentResponse {
   repeated CompactTaskAssignment task_assignment = 1;
 }
 
+message RiseCtlListCompactTaskProgressRequest {}
```
Comment (hzxa21) [inline proto/hummock.proto:702]: nits: it is better not calling it `RiseCtl...Requqest/Response` if it is not used in risectl

File: proto/hummock.proto [hunk 4]
```diff
@@ -380,6 +380,7 @@ message CompactTaskProgress {
   uint64 num_progress_key = 4;
   uint64 num_pending_read_io = 5;
   uint64 num_pending_write_io = 6;
+  uint64 compaction_group_id = 7;
```
Comment (hzxa21) [inline proto/hummock.proto:383]: optional? Compactor won't set this field when it rerpots the progress to meta.

**Review by hzxa21 — APPROVED** [review]
Rest LGTM

File: src/ctl/src/cmd_impl/hummock/compaction_group.rs (hunk 1 above)
Comment (Li0k) [reply src/ctl/src/cmd_impl/hummock/compaction_group.rs]: revert it

**Pull request merged**